code/modules/projectiles/gun_system.dm

```diff
@@ -652,6 +655,30 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 		SEND_SIGNAL(gun_user, COMSIG_MOB_ATTACHMENT_FIRED, target, src, master_gun)
 	gun_user?.client?.mouse_pointer_icon = 'icons/effects/supplypod_target.dmi'
 
+/obj/item/weapon/gun/proc/set_shoot_inactive_hand(mob/user, firing) // Handles akimbo
+	if(!user)
+		return
+	if(!dual_wield)
+		user.shoot_inactive_hand = FALSE
+		return
+	
+	var/obj/item/weapon/gun/active_gun = user.get_active_held_item()
+	var/obj/item/weapon/gun/inactive_gun = user.get_inactive_held_item()
+	if(firing) // Apply akimbo delay
+		if(user.shoot_inactive_hand)
+			active_gun.last_fired = world.time
+		else
+			inactive_gun.last_fired = world.time
+	if(user.shoot_inactive_hand && (!inactive_gun?.cell && (!inactive_gun.in_chamber?.ammo && !inactive_gun.current_mag?.current_rounds || inactive_gun.current_mag?.current_rounds && inactive_gun.current_mag.current_rounds <= 0) || inactive_gun?.cell && !inactive_gun.cell?.charge && inactive_gun.cell.charge <= 0)) // Check inactive gun
+		user.shoot_inactive_hand = FALSE // Shoot from active
+	else if(!user.shoot_inactive_hand && (!active_gun?.cell && (!active_gun.in_chamber?.ammo && !active_gun.current_mag?.current_rounds || active_gun.current_mag?.current_rounds && active_gun.current_mag.current_rounds <= 0) || active_gun?.cell && !active_gun.cell?.charge && active_gun.cell.charge <= 0)) // Check active gun
+		user.shoot_inactive_hand = TRUE // Shoot from inactive
+	else if(firing)
```

Comment:
can you explain to me when does this is usefull?

Reply:
It isn't useful, it's shitcode